torch/_export/constraints.py

```diff
@@ -0,0 +1,37 @@
+from typing import Optional
+import warnings
+
+import torch._dynamo
+from torch._dynamo.comptime import comptime
+from torch.fx.experimental.symbolic_shapes import constrain_range
+
+
+def add_inline_constraint(symbol, min: Optional[int] = None, max: Optional[int] = None):
+    """
+    Add min/max constraint on the intermediate symbol at tracing time
+    """
+
+    if torch._dynamo.is_compiling():
+        @comptime
+        def _(ctx):
+            node = ctx.get_local("symbol").as_proxy().node
+            min = ctx.get_local("min").as_python_constant()
+            max = ctx.get_local("max").as_python_constant()
+            constrain_range(node.meta["example_value"], min=min, max=max)
+            ctx.graph().call_function(constrain_range, (node,), {"min": min, "max": max})
+    else:
```

Comment:
When would this happen?

Reply:
If you just trace with make_fx then this @comptime function doesn't get called so there's no constraint added.